FreeRTOS-Plus/Demo/FreeRTOS-IoT-Libraries-LTS-Beta2/mqtt/mqtt_light_weight/DemoTasks/LightWeightMQTTExample.c

```diff
@@ -41,41 +54,938 @@
+                                    uint16_t usPacketId )
+{
+    switch( pxIncomingPacket->type )
+    {
+        case MQTT_PACKET_TYPE_SUBACK:
+            LogInfo( ( "Subscribed to the topic %s.\r\n", mqttexampleTOPIC ) );
+            /* Make sure ACK packet identifier matches with Request packet identifier. */
+            configASSERT( usSubscribePacketIdentifier == usPacketId );
+            break;
+
+        case MQTT_PACKET_TYPE_UNSUBACK:
+            LogInfo( ( "Unsubscribed from the topic %s.\r\n", mqttexampleTOPIC ) );
+            /* Make sure ACK packet identifier matches with Request packet identifier. */
+            configASSERT( usUnsubscribePacketIdentifier == usPacketId );
+            break;
+
+        case MQTT_PACKET_TYPE_PINGRESP:
+            LogInfo( ( "Ping Response successfully received.\r\n" ) );
+            break;
+
+        /* Any other packet type is invalid. */
+        default:
+            LogInfo( ( "prvMQTTProcessResponse() called with unknown packet type:(%02X).\r\n",
+                       pxIncomingPacket->type ) );
+    }
+}
+
+/*-----------------------------------------------------------*/
+
+static void prvMQTTProcessIncomingPublish( MQTTPublishInfo_t * pxPublishInfo )
+{
+    configASSERT( pxPublishInfo != NULL );
+
+    /* Process incoming Publish. */
+    LogInfo( ( "Incoming QOS : %d\n", pxPublishInfo->qos ) );
+
+    /* Verify the received publish is for the we have subscribed to. */
+    if( ( pxPublishInfo->topicNameLength == strlen( mqttexampleTOPIC ) ) &&
+        ( 0 == strncmp( mqttexampleTOPIC,
+                        pxPublishInfo->pTopicName,
+                        pxPublishInfo->topicNameLength ) ) )
+    {
+        LogInfo( ( "\r\nIncoming Publish Topic Name: %.*s matches subscribed topic.\r\n"
+                   "Incoming Publish Message : %.*s\r\n",
+                   pxPublishInfo->topicNameLength,
+                   pxPublishInfo->pTopicName,
+                   pxPublishInfo->payloadLength,
+                   pxPublishInfo->pPayload ) );
+    }
+    else
+    {
+        LogInfo( ( "Incoming Publish Topic Name: %.*s does not match subscribed topic.\r\n",
+                   pxPublishInfo->topicNameLength,
+                   pxPublishInfo->pTopicName ) );
+    }
+}
+
+/*-----------------------------------------------------------*/
+
+static void prvMQTTProcessIncomingPacket( Socket_t xMQTTSocket )
+{
+    MQTTStatus_t xResult;
+    MQTTPacketInfo_t xIncomingPacket;
+    BaseType_t xStatus;
+    MQTTPublishInfo_t xPublishInfo;
+    uint16_t usPacketId;
+    NetworkContext_t xNetworkContext;
+
+    /***
+     * For readability, error handling in this function is restricted to the use of
+     * asserts().
+     ***/
+
+    ( void ) memset( ( void * ) &xIncomingPacket, 0x00, sizeof( MQTTPacketInfo_t ) );
+
+    /* Determine incoming packet type and remaining length. */
+    xNetworkContext.xTcpSocket = xMQTTSocket;
```

Comment:
This could be 
NetworkContext_t * pNetworkContext;
pNetworkContext = &xMQTTScoket;

Reply:
The transport interface typedefs NetworkContext_t as a struct, so it can't be redefined here

Reply:
Yes, I was just wondering if can decouple it for light weight demo since it does not use full transport interface., but looks like we cannot because MQTT library is build separately and needs transport_interface declaration.